botocore/awsrequest.py

```diff
@@ -29,6 +28,11 @@
      urlencode
 from botocore.exceptions import UnseekableStreamError
 
+try:
```

Comment:
This is what I was referring to when I mentioned moving the try/except into a compat module.  It's repeated 3 times and is a perfect fit for `compat.py`.